### ISSUE TITLE
Cannot compile Modules under CentOS 7 (aarch64)

### ISSUE THREAD

**Issue by mpbaum2019** [zfsonlinux/zfs#9711]
### System information

Type | Version/Name    CentOS 7 on Raspberry Pi 4
 --- | --- 
Distribution Name |	Centos 
Distribution Version |     7.7.1908
Linux Kernel |                  4.19.86-v8+
Architecture |                  aarch64
ZFS Version |                 0.8 / master (source)
SPL Version |                 0.8 / master (source)

### Describe the problem you're observing
The `./configure` command does not complete successfully. Exits with following message:
```
checking kernel source version... 4.19.88-v8+
checking kernel file name for module symbols... Module.symvers
checking whether modules can be built... no
configure: error: 
	*** Unable to build an empty module.
```

### Describe how to reproduce the problem
Follow the instructions provided in the wiki on how to build zfs from source

### Useful information
I already tried different versions of gcc and even compiling under raspbian instead of CentOS - with the same effect. Here's the entire config.log:

[config.log.txt](https://github.com/zfsonlinux/zfs/files/3946922/config.log.txt)

**Comment by behlendorf** [comment]
From config.log.txt

> configure:38832: cp conftest.c conftest.h build && make modules -C /usr/src/kernels/4.19.86-v8+ EXTRA_CFLAGS=-Werror -Wframe-larger-than=4096   M=/root/zfs/build 
/bin/sh: scripts/basic/fixdep: cannot execute binary file

It appears that the kernel provided `/usr/src/kernels/[kernel]/scripts/basic/fixdep` binary has been compiled for the wrong architecture.  Can you double check that you've installed the kernel-devel package for the right architecture.

You can use `file` to check the binary.  For example, for an x86_64 system.

```
$ file /usr/src/kernels/3.10.0-1062.1.2.el7.x86_64/scripts/basic/fixdep
/usr/src/kernels/3.10.0-1062.1.2.el7.x86_64/scripts/basic/fixdep: ELF 64-bit LSB executable, x86-64, version 1 (SYSV), dynamically linked (uses shared libs), for GNU/Linux 2.6.32, BuildID[sha1]=419d202967252cba0ad0d13ce4187827b925a13b, stripped
```

**Comment by mpbaum** [comment]
So sorry - looks like that was the issue:
```
basic/fixdep: ELF 64-bit LSB shared object, x86-64, version 1 (SYSV), dynamically linked (uses shared libs), BuildID[sha1]=f96cf37e4ab3abdfa90880655b262c4ae72c937a, for GNU/Linux 3.2.0, not stripped```

But the kernel-devel package was compiled for aarch64; even shows that way on the system:
```
# rpm -qa | grep kernel
kernel-4.19.86_v8+-1.aarch64
kernel-devel-4.19.86_v8+-1.aarch64
kernel-headers-4.19.86_v8+-1.aarch64
´´´

Thanks for the quick reply. This is obviously not an issue with zfs, so feel free to close the ticket😊

**Comment by behlendorf** [comment]
@mpbaum no problem.  It would be great if you could file an issue with CentOS to let them know there's an issue with their aarch64 kernel-devel package.

**Comment by mpbaum2019** [comment]
Managed to fix the issue by running `make scripts` in the kernel source directory (`/usr/src/kernels/4.19.86-v8+/`)

I compiled the kernel myself using an ubuntu machine for cross-compiling - so that's on me. Still, weird that all binaries in `scripts` were x86 instead of arm64... So maybe open an issue with the raspberry-pi/linux project?

Anyway, just wanted to let you know that the `./configure` went through smoothly, running `make` right now - thanks again!
